lib/server/nahnojiresolver.py

```diff
@@ -21,10 +21,10 @@ def url(url):
 		ur = m.group('url')
 		if ur.find('.flv') > -1:
 			return [ur]
-		data = util.substr(util.request(ur),'<body style','</a>')
-		pattern = 'href=\"(.+?)\"[^>]+>' 
+		data = util.substr(util.request(ur),'<source type="video/mp4"','</source>')
+		pattern = 'src=\"(.+?)\"[^>]+>' 
```

Comment:
sadly, this is not an option as html is (purposely?) invalid as beautifulsoup is unable to process it properly. here's an example: `<html><head>..</head></html><body>..</body>`